src/colvarbias_abf.cpp

```diff
@@ -285,63 +334,22 @@ int colvarbias_abf::init(std::string const &conf)
 /// Destructor
 colvarbias_abf::~colvarbias_abf()
 {
-  if (samples) {
-    delete samples;
-    samples = NULL;
-  }
-
-  if (gradients) {
-    delete gradients;
-    gradients = NULL;
-  }
-
-  if (pmf) {
-    delete pmf;
-    pmf = NULL;
-  }
-
-  if (z_samples) {
-    delete z_samples;
-    z_samples = NULL;
-  }
-
-  if (z_gradients) {
-    delete z_gradients;
-    z_gradients = NULL;
-  }
-
-  if (czar_gradients) {
-    delete czar_gradients;
-    czar_gradients = NULL;
-  }
-
-  if (czar_pmf) {
-    delete czar_pmf;
-    czar_pmf = NULL;
-  }
-
-  // shared ABF
-  // We used to only do this if "shared" was defined,
-  // but now we can call shared externally
-  if (last_samples) {
-    delete last_samples;
-    last_samples = NULL;
-  }
-
-  if (last_gradients) {
-    delete last_gradients;
-    last_gradients = NULL;
-  }
-
-  if (system_force) {
-    delete [] system_force;
-    system_force = NULL;
+  #define delete_if_non_null(ptr) if (ptr) { delete ptr; ptr = nullptr; }
```

Comment:
It could be better if `nullptr` and `NULL` can be unified.

Reply:
I agree, now there is a mix but the NULL's could be changed to nullptr. @giacomofiorin , do you agree?

Reply:
@jhenin Yes, totally. I just don't bring this up explicitly because it's quite straightforward to run [`clang-tidy`](https://clang.llvm.org/extra/clang-tidy/checks/modernize/use-nullptr.html).

Even better to use smart pointers, though, in which case the `NULL` could server as a reminder ;-)